### ISSUE TITLE
mpdf 7.x: absolute-bottom-positioned elements not working

### ISSUE THREAD

**Issue by heldchen** [mpdf/mpdf#304]
in the develop branch, elements that are absolute "bottom" positioned are not really positioned from the bottom of the page, but at the top (example screenshot: red). "top"-aligned absolute elements (example screenshot: green) work fine.

works fine in release 6.1.3

### These are mPDF and PHP versions I am using

mpdf: `dev-development` (composer)
php: tested with 7.0.14 & 7.1.0 (debian)

### This is a HTML code snippet I use

```
<html>
<head>
    <style>
        .target1 {
            position:absolute;
            bottom:2.15cm;
            left:3.65cm;
            right:2.15cm;
            background-color: red;
        }
        .target2 {
            position:absolute;
            top:5.15cm;
            left:3.65cm;
            right:2.15cm;
            background-color: green;
        }

    </style>
</head>
<body>
    <div class="target1">hello</div>
    <div class="target2">world</div>
</body>
</html>
```

### This is the result
![image](https://cloud.githubusercontent.com/assets/299636/21826908/9c8037a8-d789-11e6-8ad4-f1dbf9600268.png)

**Comment by heldchen** [comment]
the problem was introduced with the php 7.1 fixes, specifically on the lines 16509-16512: 
```
			$bbox_top = is_numeric($bbox_top) ? $bbox_top : 0;
			$bbox_left = is_numeric($bbox_left) ? $bbox_left : 0;
			$bbox_right = is_numeric($bbox_right) ? $bbox_right : 0;
			$bbox_bottom = is_numeric($bbox_bottom) ? $bbox_bottom : 0;
```

up until this lines, `$bbox_top` contains auto and `$bbox_bottom` the converted css value. now `$bbox_top` is falsely set to `0` because it is not numeric. it should be set to `$bbox_bottom - $h`, yet the height is not yet calculated...

a few lines further down `$bbox_top` is specifically checked for `auto`, but currently this will never be this value.

**Comment by heldchen** [comment]
this fixes the problem, but the temp variables are rather ugly:
```
			$bbox_ntop = is_numeric($bbox_top) ? $bbox_top : 0;
			$bbox_nleft = is_numeric($bbox_left) ? $bbox_left : 0;
			$bbox_nright = is_numeric($bbox_right) ? $bbox_right : 0;
			$bbox_nbottom = is_numeric($bbox_bottom) ? $bbox_bottom : 0;

			$y = $cont_y + $bbox_ntop + $bbox_mt + $bbox_bt + $bbox_pt;
			$h = $cont_h - $bbox_ntop - $bbox_mt - $bbox_bt - $bbox_pt - $bbox_pb - $bbox_bb - $bbox_mb - $bbox_nbottom;

			$x = $cont_x + $bbox_nleft + $bbox_ml + $bbox_bl + $bbox_pl;
			$w = $cont_w - $bbox_nleft - $bbox_ml - $bbox_bl - $bbox_pl - $bbox_pr - $bbox_br - $bbox_mr - $bbox_nright;

			// Set (temporary) values for x y w h to do first paint, if values are auto
			if ($inner_h === 'auto' && $bbox_top === 'auto') {
				$y = $cont_y + $bbox_mt + $bbox_bt + $bbox_pt;
				$h = $cont_h - ($bbox_nbottom + $bbox_mt + $bbox_mb + $bbox_bt + $bbox_bb + $bbox_pt + $bbox_pb);
			} elseif ($inner_h === 'auto' && $bbox_bottom === 'auto') {
				$y = $cont_y + $bbox_ntop + $bbox_mt + $bbox_bt + $bbox_pt;
				$h = $cont_h - ($bbox_ntop + $bbox_mt + $bbox_mb + $bbox_bt + $bbox_bb + $bbox_pt + $bbox_pb);
			}
			if ($inner_w === 'auto' && $bbox_left === 'auto') {
				$x = $cont_x + $bbox_ml + $bbox_bl + $bbox_pl;
				$w = $cont_w - ($bbox_nright + $bbox_ml + $bbox_mr + $bbox_bl + $bbox_br + $bbox_pl + $bbox_pr);
			} elseif ($inner_w === 'auto' && $bbox_right === 'auto') {
				$x = $cont_x + $bbox_nleft + $bbox_ml + $bbox_bl + $bbox_pl;
				$w = $cont_w - ($bbox_nleft + $bbox_ml + $bbox_mr + $bbox_bl + $bbox_br + $bbox_pl + $bbox_pr);
			}
			$bbox_y = $cont_y + $bbox_ntop + $bbox_mt;
			$bbox_x = $cont_x + $bbox_nleft + $bbox_ml;
```

**Comment by jbrooksuk** [comment]
@finwe any chance we can see a fix for this?

**Comment by finwe** [comment]
@jbrooksuk With no ETA.

**Comment by finwe** [comment]
Fixed in e55032a

**Comment by jbrooksuk** [comment]
Whoop!
